### PR TITLE
New page list architecture for Teacher

### PR DESCRIPTION
**Pull request by GabrielPalmer**

### CODE REVIEW

File: Core/CoreTests/Pages/PageList/PageListPresenterTests.swift [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        expectation.assertForOverFulfill = false
+        onUpdate = {
+            if !self.coursePresenter.pages.isEmpty {
+                expectation.fulfill()
+
+            }
+        }
+        coursePresenter.viewIsReady()
+        wait(for: [expectation], timeout: 1)
+        XCTAssertEqual(coursePresenter.pages.first?.title, "Answers Page")
```
Comment (armstrongnate) [inline Core/CoreTests/Pages/PageList/PageListPresenterTests.swift]: This test was flaky when it was in student. I think I fixed in in this PR https://github.com/instructure/canvas-ios/pull/472

Would you mind making that change here?

File: rn/Teacher/index.ios.js [hunk 2]
```diff
@@ -123,11 +123,13 @@ const loginHandler = async ({
 
   setSession(session)
 
-  try {
-    await getUser('self')
-  } catch (err) {
-    if (err.response && err.response.status === 401) {
-      return NativeLogin.logout()
+  if (!NativeModules.SettingsManager.settings.IS_UI_TEST) {
```
Comment (armstrongnate) [inline rn/Teacher/index.ios.js]: This was probably a bad merge. We don't want this check anymore.

File: rn/Teacher/src/common/login-verify.js [hunk 3]
```diff
@@ -24,6 +24,9 @@ import canvas from '../canvas-api'
 // if the user has an invalid login, the promise will send `true`. Otherwise it will send `false`
 export default async function loginVerify (): Promise<boolean> {
   return new Promise((resolve, reject) => {
+    if (NativeModules.SettingsManager.settings.IS_UI_TEST) {
+      return resolve(false)
+    }
```
Comment (armstrongnate) [inline rn/Teacher/src/common/login-verify.js]: Also a bad merge. You can remove these additions.

**Comment by armstrongnate** [comment]
@GabrielPalmer You probably need to merge in master. If you need help resolving the conflicts let me know.

Once that's done I'll QA from the new build. Code looks good though.

**Comment by GabrielPalmer** [comment]
@armstrongnate Yeah, I'm going to need your help. Those look like some big changes to the file directories.

**Pull request merged**